packages/backend/src/plugins/extensions/auth.ts

```diff
@@ -76,13 +76,20 @@ export const authModuleMicrosoftProvider = createBackendModule({
                         authenticator: microsoftAuthenticator,
                         async signInResolver(info, ctx) {
                             const catalogApi = new CatalogClient({ discoveryApi: discovery })
-                            const email = info.profile.email
-                            if (!email) {
-                                throw new AuthenticationError('Authentication failed', "No username found in profile");
+                            const { result } = info
+
+                            if (!result.session.accessToken) {
+                                throw new Error(
+                                    "Login failed, OAuth session did not contain an access token",
+                                )
                             }
+
+                            const oid = getObjectIdFromToken(
+                                result.session.accessToken,
+                            )
                             const { entity } = await ctx.findCatalogUser({
                                 annotations: {
-                                    'microsoft.com/email': email
+                                    "graph.microsoft.com/user-id": oid,
```

Comment:
Does this still work with local development using the `org.yaml` test file? 
Is it still possible to use Github Apps for authentication in combination with `org.yaml` in local development?

Reply:
It will work with local development using the `org.yaml` test file as the same annotation is present there. I have not tested with using a GitHub App for authentication, but the signin resolver defined in `authModuleGithubLocalProvider` will handle this case.

Reply:
Okey. Maybe we should add a little comment in the CONTRIBUTING.md file on how to test locally with azure?